### PR TITLE
Paul changes

### PR DESCRIPTION
**Pull request by pkenjora**
I've added a configuration object that replaces input_parameters.  This allows three things:

1.  A Main that takes command line arguments (partial parameters as its a minimal interface).
2. A Colab friendly object that can be mutated across cells.
3. Future proofed a bit as values can be added to the Configuration without affecting function parameters.

### CODE REVIEW

**Review by anaesqueda — APPROVED** [review]
LGTM, we will start adding tests that can cover these changes.

**Review by brugz — APPROVED** [review]
LGTM for Python

**Review by brugz — APPROVED** [review]
LGTM for Python

File: configuration.py [hunk 1]
```diff
@@ -32,42 +32,25 @@
...
-        self.bucket_name = bucket_name
-        self.knowledge_graph_api_key = knowledge_graph_api_key
-        self.use_annotations = use_annotations
-        self.use_llms = use_llms
-        self.assessment_file = assessment_file
-        self.verbose = verbose
-
-        self.annotation_path = f"gs://{bucket_name}/ABCD/"
+        # set parameters
+        self.project_id = ""
```
Comment (brugz) [inline configuration.py:43]: You should try to differentiate public v. private attributes w/ underscore (`_`) as necessary.

**Review by brugz — APPROVED** [review]
LGTM for Python

File: configuration.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+###########################################################################
+
+"""Module that defines global parameters"""
+
+import os
+
+FFMPEG_BUFFER = "reduced/buffer.mp4"
```
Comment (brugz) [inline configuration.py:25]: Are these intended to be public?

File: configuration.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        self.top_k = 32
+
+
+    def set_parameters(self,
+        project_id: str,
+        project_zone: str,
+        bucket_name: str,
+        knowledge_graph_api_key: str,
+        bigquery_dataset:str,
+        bigquery_table:str,
```
Comment (brugz) [inline configuration.py:88]: Check whitespace on the type annotations

**Review by brugz — APPROVED** [review]
LGTM for Python

File: configuration.py (hunk 1 above)
Comment (pkenjora) [reply configuration.py:43]: Yes, in this case they are all public.

File: configuration.py (hunk 2 above)
Comment (pkenjora) [reply configuration.py:25]: Yes, we left them out of the class because they are typically configured at the application layer not at instantiation.

**Pull request merged**